### PR TITLE
[sitecore-jss-nextjs] Fix page not found error on pages.

### PR DESCRIPTION
**Pull request by sc-addypathania**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
When switching between page and subpage on Pages we get Page not found error because there are duplicate cookies set by nextjs `setPreviewData` function on each page route. There should be single active preview cookie at the root of your session.

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by ambrauer — APPROVED** [review]
Looks good. Is there some unit test that will need updating (or should be added)?

**Comment by sc-addypathania** [comment]
> Looks good. Is there some unit test that will need updating (or should be added)?

Nothing really to update in tests since it was a optional options param in the `setPreviewData` function.

File: packages/sitecore-jss-nextjs/src/editing/editing-render-middleware.test.ts [hunk 1]
```diff
@@ -70,9 +70,6 @@ const mockResponse = () => {
   res.clearPreviewData = spy(() => {
     return res;
   });
-  res.setPreviewData = spy(() => {
```
Comment (art-alexeyenko) [inline packages/sitecore-jss-nextjs/src/editing/editing-render-middleware.test.ts:73]: This spy is still needed

Comment (sc-addypathania) [reply packages/sitecore-jss-nextjs/src/editing/editing-render-middleware.test.ts:73]: Yeah yeah i accidently removed the `setPreviewData`.
I wanted to remove the `clearPreviewData`.

**Pull request merged**